In [9]:
import pandas as pd
import numpy as np

In [10]:
file_path = "../data/raw/airbnb_listing.csv"

df = pd.read_csv(file_path)
df.head()

,host_response_rate,host_total_listings_count,review_scores_accuracy,host_acceptance_rate,availability_90,neighbourhood_cleansed,availability_60,bedrooms,review_scores_cleanliness,host_response_time,...,host_since,number_of_reviews,review_scores_location,price,review_scores_communication,number_of_reviews_ltm,review_scores_value,availability_30,beds,availability_365
0,NaN,NaN,3.44,NaN,46.0,THIRD WARD,16.0,2.0,3.56,NaN,...,NaN,9.0,3.22,NaN,4.56,0.0,3.67,0.0,2.0,321.0
1,NaN,NaN,4.88,NaN,26.0,SIXTH WARD,0.0,1.0,4.87,NaN,...,NaN,317.0,4.82,NaN,4.81,10.0,4.78,0.0,1.0,301.0
2,NaN,NaN,4.61,NaN,35.0,SECOND WARD,31.0,0.0,4.45,NaN,...,NaN,389.0,4.76,NaN,4.86,20.0,4.63,18.0,2.0,35.0
3,NaN,NaN,NaN,NaN,71.0,THIRTEENTH WARD,45.0,3.0,NaN,NaN,...,NaN,0.0,NaN,NaN,NaN,0.0,NaN,20.0,3.0,324.0
4,NaN,NaN,4.83,NaN,73.0,SIXTH WARD,43.0,1.0,4.71,NaN,...,NaN,332.0,4.80,NaN,4.70,1.0,4.72,13.0,1.0,348.0


In [11]:
df.shape

(42904, 38)

In [12]:
df.columns.tolist()

['host_response_rate',
 'host_total_listings_count',
 'review_scores_accuracy',
 'host_acceptance_rate',
 'availability_90',
 'neighbourhood_cleansed',
 'availability_60',
 'bedrooms',
 'review_scores_cleanliness',
 'host_response_time',
 'reviews_per_month',
 'amenities',
 'host_listings_count',
 'longitude',
 'room_type',
 'bathrooms',
 'review_scores_rating',
 'estimated_occupancy_l365d',
 'estimated_revenue_l365d',
 'property_type',
 'number_of_reviews_ly',
 'review_scores_checkin',
 'accommodates',
 'latitude',
 'maximum_nights',
 'instant_bookable',
 'host_is_superhost',
 'minimum_nights',
 'host_since',
 'number_of_reviews',
 'review_scores_location',
 'price',
 'review_scores_communication',
 'number_of_reviews_ltm',
 'review_scores_value',
 'availability_30',
 'beds',
 'availability_365']

In [13]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 42904 entries, 0 to 42903
Data columns (total 38 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   host_response_rate           32295 non-null  str    
 1   host_total_listings_count    42424 non-null  float64
 2   review_scores_accuracy       35784 non-null  float64
 3   host_acceptance_rate         35340 non-null  str    
 4   availability_90              42902 non-null  float64
 5   neighbourhood_cleansed       42902 non-null  str    
 6   availability_60              42902 non-null  float64
 7   bedrooms                     41363 non-null  float64
 8   review_scores_cleanliness    35781 non-null  float64
 9   host_response_time           32294 non-null  str    
 10  reviews_per_month            35783 non-null  float64
 11  amenities                    42902 non-null  str    
 12  host_listings_count          42876 non-null  float64
 13  longitude                  

In [14]:
df.describe()

,host_total_listings_count,review_scores_accuracy,availability_90,availability_60,bedrooms,review_scores_cleanliness,reviews_per_month,host_listings_count,longitude,bathrooms,...,maximum_nights,minimum_nights,number_of_reviews,review_scores_location,review_scores_communication,number_of_reviews_ltm,review_scores_value,availability_30,beds,availability_365
count,42424.000000,3.578400e+04,42902.000000,42902.000000,41363.000000,35781.000000,35783.000000,42876.000000,42902.000000,33090.000000,...,42902.000000,42902.000000,42902.000000,35779.000000,35782.000000,42902.000000,35778.000000,42902.000000,32976.000000,42901.000000
mean,34.425726,7.977322e+01,45.216237,27.189875,1.506298,4.756458,1.357060,23.472805,29.631573,1.300620,...,461.470491,7.751527,52.862827,4.726750,4.857957,10.251410,4.691585,11.109249,2.038483,195.300179
std,164.376987,1.418223e+04,34.015698,22.122010,1.274222,0.377259,1.940987,87.187957,49.407679,0.761264,...,806.064095,33.318036,109.404180,0.375201,0.322518,21.362045,0.384735,10.771320,1.789059,139.875129
min,0.000000,1.000000e+00,0.000000,0.000000,0.000000,0.000000,0.010000,0.000000,-82.669890,0.000000,...,1.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,1.000000,4.750000e+00,1.000000,0.000000,1.000000,4.680000,0.240000,1.000000,4.884932,1.000000,...,50.000000,1.000000,2.000000,4.640000,4.850000,0.000000,4.600000,0.000000,1.000000,39.000000
50%,4.000000,4.900000e+00,54.000000,28.000000,1.000000,4.870000,0.660000,3.000000,23.725932,1.000000,...,365.000000,2.000000,13.000000,4.830000,4.960000,2.000000,4.780000,8.000000,2.000000,234.000000
75%,16.000000,5.000000e+00,76.000000,48.000000,2.000000,5.000000,1.850000,13.000000,100.467605,1.500000,...,1125.000000,3.000000,53.000000,4.970000,5.000000,11.000000,4.910000,21.000000,2.000000,334.000000
max,8774.000000,2.682808e+06,90.000000,60.000000,50.000000,5.000000,99.420000,1649.000000,100.923710,22.000000,...,100000.000000,1115.000000,5097.000000,5.000000,5.000000,1304.000000,5.000000,30.000000,50.000000,365.000000


In [15]:
# count missing values in each column

df.isnull().sum().sort_values(ascending=False)

host_response_time             10610
host_response_rate             10609
estimated_revenue_l365d        10319
price                          10319
beds                            9928
bathrooms                       9814
host_acceptance_rate            7564
review_scores_value             7126
review_scores_location          7125
review_scores_checkin           7124
review_scores_cleanliness       7123
review_scores_communication     7122
review_scores_rating            7121
reviews_per_month               7121
review_scores_accuracy          7120
host_is_superhost               1575
bedrooms                        1541
host_since                       480
host_total_listings_count        480
instant_bookable                 455
host_listings_count               28
availability_365                   3
accommodates                       2
latitude                           2
maximum_nights                     2
number_of_reviews_ly               2
estimated_occupancy_l365d          2
m

In [16]:
df.duplicated().sum()

np.int64(52)

## Column Categorization

### 1. Target Variables (KPIs)
- estimated_revenue_l365d
- estimated_occupancy_l365d

### 2. Key Features for Analysis
- room_type
- price
- accommodates
- bedrooms, bathrooms, beds
- amenities
- neighbourhood_cleansed
- host_is_superhost

### 3. Columns with High Missing Values
- host_neighbourhood
- neighborhood_overview
- host_about
- license

### 4. Columns to Drop (Completely Missing)
- neighbourhood_group_cleansed
- calendar_updated

### 5. Columns Requiring Type Conversion
- price (object → float)
- host_response_rate (% → float)
- host_acceptance_rate (% → float)
- date columns → datetime

## Conclusion

- Dataset contains 42,903 listings with 79 features
- No duplicate rows found
- Several columns have missing values, including some completely empty columns
- Key variables for business problem identified
- Data type inconsistencies observed in price and percentage fields

